### PR TITLE
AB#88277 Add integration test for delivery Part I

### PR DESCRIPTION
**Pull request by dawid-pawulski**

### CODE REVIEW

File: src/HE.Investment.AHP.WWW/Views/DeliveryPhase/BuildActivityType.cshtml [hunk 1]
```diff
@@ -32,7 +32,7 @@
 }
 
 <gds-div-grid-column-two-thirds>
-    <vc:ahp-workflow-back-button current-page="@DeliveryPhaseWorkflowState.BuildActivityType"/>
+    <vc:ahp-workflow-back-button current-page="@DeliveryPhaseWorkflowState.TypeOfHomes"/>
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.WWW/Views/DeliveryPhase/BuildActivityType.cshtml:35]: this change looks wrong, this is BuildActivityType page

File: src/HE.Investment.AHP.WWW/Views/DeliveryPhase/RemoveDeliveryPhaseConfirmation.cshtml [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 @using HE.Investment.AHP.WWW.Models.Delivery
+@using HE.Investment.AHP.WWW.Views.Delivery.Const
+@using HE.Investment.AHP.WWW.Views.Shared.Components.ReturnToApplicationLink
 @using HE.Investments.Common.WWW.Components.BackLinkWithCallback
 @using HE.Investments.Common.WWW.Components.FormHeader
 @using HE.Investments.Common.WWW.Components.FormButton
 @model HE.Investment.AHP.WWW.Models.Delivery.RemoveDeliveryPhaseModel;
 @{
     var applicationId = this.GetApplicationIdFromRoute();
+    var title = DeliveryPageTitles.Remove;
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.WWW/Views/DeliveryPhase/RemoveDeliveryPhaseConfirmation.cshtml:11]: add title to `ViewData` as well

File: src/HE.Investments.AHP.IntegrationTests/FillApplication/Order05CompleteDeliveryPhases.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        // then
+        newDeliveryPhasePage
+            .UrlEndWith(BuildDeliveryPhasesPage(DeliveryPhasesPagesUrl.NewDeliveryPhase))
+            .HasTitle(DeliveryPageTitles.Name)
+            .HasGdsSubmitButton("continue-button", out var continueButton);
+
+        var deliveryPhase = NewBuildAndWorksOnlyDeliveryPhase.GenerateDeliveryPhase();
+        var deliveryPhaseNamePage = await TestClient.SubmitButton(continueButton, ("DeliveryPhaseName", deliveryPhase.Name.ToString()!));
+
+        NewBuildAndWorksOnlyDeliveryPhase.SetDeliveryPhaseId(deliveryPhaseNamePage.Url.GetHomeTypeGuidFromUrl());
```
Comment (jjankowski87) [inline src/HE.Investments.AHP.IntegrationTests/FillApplication/Order05CompleteDeliveryPhases.cs:78]: it's not HomeType guid :)

File: src/HE.Investments.AHP.IntegrationTests/FillApplication/Data/NewBuildAndWorksOnlyDeliveryPhase.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    public CompletionMilestoneDetails CompletionMilestone { get; private set; }
+
+    public void SetDeliveryPhaseId(string deliveryPhaseId)
+    {
+        Id = deliveryPhaseId;
+    }
+
+    public NewBuildAndWorksOnlyDeliveryPhase GenerateDeliveryPhase()
+    {
+        Name = new DeliveryPhaseName($"IT-General-{DateTime.UtcNow.ToString(CultureInfo.InvariantCulture)}");
```
Comment (jjankowski87) [inline src/HE.Investments.AHP.IntegrationTests/FillApplication/Data/NewBuildAndWorksOnlyDeliveryPhase.cs:31]: Not General :)

**Review by jjankowski87 — APPROVED** [review]

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Views/Delivery/Index.cshtml [hunk 5]
```diff
@@ -33,7 +33,7 @@
         <gds-p>If your application consist of multiple activities, set up separate delivery phases to cover them. Do this if you are delivering a mix of new build and rehab, or delivering the homes on this application in phases.</gds-p>
     </div>
 
-    <a href="@Url.Action("List", "Delivery", new { applicationId = Model.Id.Value })">
+    <a href="@Url.Action("List", "Delivery", new { applicationId = Model.Id.Value })" id="continue-button">
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Views/Delivery/Index.cshtml:36]: According ADR you should not use ids.

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Views/DeliveryPhase/RemoveDeliveryPhaseConfirmation.cshtml (hunk 2 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.WWW/Views/DeliveryPhase/RemoveDeliveryPhaseConfirmation.cshtml:11]: done

File: src/HE.Investments.AHP.IntegrationTests/FillApplication/Data/NewBuildAndWorksOnlyDeliveryPhase.cs (hunk 4 above)
Comment (dawid-pawulski) [reply src/HE.Investments.AHP.IntegrationTests/FillApplication/Data/NewBuildAndWorksOnlyDeliveryPhase.cs]: fixed ;)

File: src/HE.Investments.AHP.IntegrationTests/FillApplication/Order05CompleteDeliveryPhases.cs (hunk 3 above)
Comment (dawid-pawulski) [reply src/HE.Investments.AHP.IntegrationTests/FillApplication/Order05CompleteDeliveryPhases.cs]: The method for delivery phase will look the same so I have changed the name of the method to: GetNestedGuidFromUrl

File: src/HE.Investment.AHP.WWW/Views/Delivery/Index.cshtml (hunk 5 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.WWW/Views/Delivery/Index.cshtml]: changed to data-testid

**Pull request merged**